libbeat/docs/shared-autodiscover.asciidoc

```diff
@@ -203,39 +203,173 @@ Example:
+These are the fields available within config templating. The `kubernetes.*` fields will be available on each emitted event:
 
 [float]
 ====== Generic fields:
   * host
   * port (if exposed)
-  * kubernetes.labels
-  * kubernetes.annotations
 
 [float]
 ====== Pod specific:
-  * kubernetes.container.id
-  * kubernetes.container.image
-  * kubernetes.container.name
-  * kubernetes.namespace
-  * kubernetes.node.name
-  * kubernetes.pod.name
-  * kubernetes.pod.uid
+|===
+|Key |Type |Description
+
+|`kubernetes.namespace`
+|`string`
+|Namespace, where the Pod is running
+
+|`kubernetes.namespace_uuid`
+|`string`
+|UUID of the Namespace, where the Pod is running
+
+|`kubernetes.namespace_labels.*`
+|`object`
+|Labels of the Namespace, where the Pod is running. All namespace labels are available by default, see `add_resource_metadata` configuration setting above.
+
+|`kubernetes.namespace_annotations.*`
+|`object`
+|Annotations of the Namespace, where the Pod is running. Namespace annotations are not available by default, see `add_resource_metadata` configuration setting above.
+
+|`kubernetes.pod.name`
+|`string`
+|Name of the Pod
+
+|`kubernetes.pod.uuid`
+|`string`
+|UUID of the Pod
+
+|`kubernetes.pod.ip`
+|`string`
+|IP of the Pod
+
+|`kubernetes.labels.*`
+|`object`
+|Object of the Pod labels. All labels are available by default.
+
+|`kubernetes.annotations.*`
+|`object`
+|Object of the Pod annotations. Annotations are not available by default.
+
+|`kubernetes.container.name`
+|`string`
+|Name of the container
+
+|`kubernetes.container.runtime`
+|`string`
+|Runtime of the container
+
+|`kubernetes.container.id`
+|`string`
+|ID of the container
+
+|`kubernetes.container.image`
+|`string`
+|Image of the container
+
+|`kubernetes.container.port`
+|`string`
+|Port of the container (if defined)
+
+|`kubernetes.container.port_name`
+|`string`
+|Port's name for the container (if defined)
+
+|`kubernetes.node.name`
+|`string`
+|Name of the Node
+
+|`kubernetes.node.uid`
+|`string`
+|UID of the Node
+
+|`kubernetes.node.hostname`
+|`string`
+|Hostname of the Node
+
+|`kubernetes.node.labels.*`
+|`string`
+|Labels of the Node. All node labels are available by default, see `add_resource_metadata` configuration setting above
+
+|`kubernetes.node.annotations.*`
+|`string`
+|Annotations of the Node. Node annotations are not available by default, see `add_resource_metadata` configuration setting above
+
+|`kubernetes.deployment.name.*`
+|`string`
+|Deployment name of the Pod (if exists). Deployment name is not available by default, see `add_resource_metadata` configuration setting above
+|===
 
 [float]
 ====== Node specific:
-  * kubernetes.node.name
-  * kubernetes.node.uid
+|===
+|Key |Type |Description
+
+|`kubernetes.labels.*`
+|`object`
+|Object of labels of the Node. All labels are available by default
+
+|`kubernetes.annotations.*`
+|`object`
+|Object of labels of the Node. Annotations are not available by default
```

Comment:
There seems to be a typo here:
```suggestion
|Object of annotations of the Node. Annotations are not available by default
```

Reply:
done in this commit https://github.com/elastic/beats/pull/37578/commits/2fbf5457f2a3855a08a3d817cafa0b97d7cb9603